src/coreComponents/constitutive/ConstitutiveManager.cpp

```diff
@@ -55,33 +55,59 @@ void ConstitutiveManager::expandObjectCatalogs()
 }
 
 
-ConstitutiveBase &
+void
 ConstitutiveManager::hangConstitutiveRelation( string const & constitutiveRelationInstanceName,
                                                dataRepository::Group * const parent,
                                                localIndex const numConstitutivePointsPerParentIndex ) const
 {
-  ConstitutiveBase const & constitutiveRelation = getConstitutiveRelation( constitutiveRelationInstanceName );
-
-  std::unique_ptr< ConstitutiveBase > material = constitutiveRelation.deliverClone( constitutiveRelationInstanceName, parent );
-
-  material->allocateConstitutiveData( *parent,
-                                      numConstitutivePointsPerParentIndex );
-
   dataRepository::Group * constitutiveGroup = parent->getGroupPointer( groupKeyStruct::constitutiveModelsString() );
-  if( constitutiveGroup == nullptr )
+  if( constitutiveGroup == nullptr )   // Why do we need to do this check? shouldn't it always be there?
```

Comment:
The way we currently use this by passing a descendant of `ElementSubRegionBase` - yes, it always has a constitutive subgroup in it. Bit the function signature takes an arbitrary `Group`, which may or may not have it. If we dropped the check, someone could unwittingly try to put constitutive models on something that is not a subregion and get segfaults. We could change the function signature to take `ElementSubRegionBase`, but that would require taking a dependency on `mesh` which I believe is not desirable and possibly circular.